MailContainerTest/Services/MailTransferService.cs

```diff
@@ -1,93 +1,77 @@
 ﻿using MailContainerTest.Data;
+using MailContainerTest.Providers;
 using MailContainerTest.Types;
+using Microsoft.Extensions.Configuration;
 using System.Configuration;
 
 namespace MailContainerTest.Services
 {
     public class MailTransferService : IMailTransferService
     {
-        public MakeMailTransferResult MakeMailTransfer(MakeMailTransferRequest request)
+        private IMailContainerStoreProvider _mailContainerStoreProvider;
+        private IMailContainerDataStore _mailContainerDataStore;
+
+        public MailTransferService(IMailContainerStoreProvider mailContainerStoreProvider)
         {
-            var dataStoreType = ConfigurationManager.AppSettings["DataStoreType"];
+            _mailContainerStoreProvider = mailContainerStoreProvider;
+
+            _mailContainerDataStore = _mailContainerStoreProvider.GetDataStore();
+        }
 
-            MailContainer mailContainer = null;
+        public MakeMailTransferResult MakeMailTransfer(MakeMailTransferRequest request)
+        {
+            var result = new MakeMailTransferResult();
 
-            if (dataStoreType == "Backup")
+            try
             {
-                var mailContainerDataStore = new BackupMailContainerDataStore();
-                mailContainer = mailContainerDataStore.GetMailContainer(request.SourceMailContainerNumber);
+                var sourceMailContainer = _mailContainerDataStore?.GetMailContainer(request.SourceMailContainerNumber);
 
-            } else
+                var destinationMailContainer = _mailContainerDataStore?.GetMailContainer(request.DestinationMailContainerNumber);
+
+                if (sourceMailContainer != null
+                    && destinationMailContainer != null
+                    && sourceMailContainer.AllowedMailType.HasFlag(request.MailType)
+                    && CanPerformTransferBetweenContainers(sourceMailContainer, destinationMailContainer))
+                {
+                    var transferSucessful = PerformTransfer(request.NumberOfMailItems, sourceMailContainer, destinationMailContainer);
```

Comment:
Plan was to have PerformTransfer return a complex object not only with a bool to signify whether it was successful, but also an error as to why it wasn't successful, so this could be logged or returned to the user